lib/assets/javascripts/src/local-time/element_observations.coffee

```diff
@@ -0,0 +1,52 @@
+import LocalTime from "./local_time"
+
+{elementMatchesSelector} = LocalTime
+
+class LocalTime.ElementObservations
+  OBSERVABLE_ATTRIBUTES = [ "datetime", "data-local", "data-format", "data-format24" ]
+
+  constructor: (@selector, @callback) ->
+    @elements = new Map()
+
+  include: (element) =>
+    unless @elements.get(element)
+      observer = @observe(element)
+      @register(element, observer)
+
+  disregard: (element) =>
+    if observer = @elements.get(element)?.observer
+      observer.disconnect()
+      @elements.delete(element)
+
+  observe: (element) =>
+    observer = new MutationObserver(@processMutations)
+    observer.observe(element, characterData: true, subtree: true, attributes: true, attributeFilter: OBSERVABLE_ATTRIBUTES)
```

Comment:
Per [Mozilla](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent), setting `textContent` (which is what this library uses to update the element) _removes all of the node's children and replaces them with a single text node with the given string value_. So it shouldn't cause infinite loops due to this configuration.

This was also my experience while testing. But of course we'll want to thoroughly test on our apps before a public release.